geonode/base/api/views.py

```diff
@@ -1415,6 +1419,85 @@ def linked_resources(self, request, pk, *args, **kwargs):
 
         return base_linked_resources(self.get_object().get_real_instance(), request.user, request.GET)
 
+    @action(
+        detail=True,
+        methods=["post"],
+        url_path=r"assets",
+        permission_classes=[UserHasPerms(perms_dict={"default": {"POST": ["base.add_resourcebase"]}})],
+        parser_classes=[JSONParser, MultiPartParser],
+        url_name="assets",
+    )
+    def asset(self, request, pk=None, *args, **kwargs):
+        """Handles uploading a new file, creating an asset, and linking it to the resource."""
+
+        if "file" not in request.FILES:
+            return Response({"message": "File not provided."}, status=status.HTTP_400_BAD_REQUEST)
+        file = request.FILES["file"]
```

Comment:
The `asset` itself can be composed by multiple files, here we should manage the file as list rather than single file, later in the code, we should evaluate that all the files uploaded are accepted by the settings